client/mm/libxc/coinbase.go

```diff
@@ -0,0 +1,1679 @@
+	}
+
+	acctID, err := c.getAccountID(ctx, ticker)
+	if err != nil {
+		return nil, err
+	}
+
+	nextURI := fmt.Sprintf("v2/accounts/%s/transactions", acctID)
+	for i := 0; i < 100 && nextURI != ""; i++ {
+		var res cbtypes.ListTransactionsResponse
+		if err := c.request(ctx, http.MethodGet, nextURI, nil, nil, &res); err != nil {
+			return nil, fmt.Errorf("error fetching transactions: %w", err)
+		}
+
+		for _, tx := range res.Data {
+			if tx.Network.Hash == txID {
+				return tx, nil
+			}
+		}
+
+		if len(res.Data) > 0 {
+			date := res.Data[len(res.Data)-1].CreatedAt
+			if date.Before(time.Now().Add(-time.Hour * 24 * 3)) {
+				return nil, fmt.Errorf("transaction %s not found", txID)
+			}
+		}
+
+		nextURI = res.Pagination.NextURI
+		nextURI = strings.TrimPrefix(nextURI, "/")
+	}
+
+	return nil, fmt.Errorf("transaction %s not found", txID)
+}
+
+// ConfirmDeposit checks if a deposit has been confirmed and returns the
+// amount deposited. This will only work for deposits that have been
+// in the past 3 days.
+func (c *coinbase) ConfirmDeposit(ctx context.Context, deposit *DepositData) (bool, uint64) {
+	tx, err := c.getTransactionByTxID(ctx, deposit.AssetID, deposit.TxID)
+	if err != nil {
+		c.log.Errorf("error fetching transaction: %v", err)
+		return false, 0
+	}
+
+	if tx.Status != "completed" {
+		return false, 0
+	}
+
+	ui, err := asset.UnitInfo(deposit.AssetID)
+	if err != nil {
+		c.log.Errorf("error getting unit info for asset ID %d: %v", deposit.AssetID, err)
+		return false, 0
+	}
+
+	return true, toAtomic(tx.Amount.Amount, &ui)
+}
+
+// Withdraw sends funds to the specified address. Coinbase adds the fee on top
+// of the amount, so more than the amount specified will be deducted from the
+// balance.
+func (c *coinbase) Withdraw(ctx context.Context, assetID uint32, amt uint64, address string) (string, uint64, error) {
+	ticker, found := c.idTicker[assetID]
+	if !found {
+		return "", 0, fmt.Errorf("no ticker found for asset ID %d", assetID)
+	}
+
+	acctID, err := c.getAccountID(ctx, ticker)
+	if err != nil {
+		return "", 0, err
+	}
+
+	assets := c.assets.Load().(map[string]*coinbaseAsset)
+	cbAsset, found := assets[ticker]
+	if !found {
+		return "", 0, fmt.Errorf("asset %s not found", ticker)
+	}
+
+	ui, err := asset.UnitInfo(assetID)
+	if err != nil {
+		return "", 0, fmt.Errorf("error getting unit info for asset ID %d: %v", assetID, err)
+	}
+
+	req := cbtypes.SendTransactionRequest{
+		Type:     "send",
+		To:       address,
+		Amount:   strconv.FormatFloat(toConv(amt, &ui), 'f', cbAsset.Exponent, 64),
+		Currency: cbAsset.Code,
+	}
+
+	if token := asset.TokenInfo(assetID); token != nil {
+		if token.ParentID == 60 {
+			req.Network = "ethereum"
+		} else if token.ParentID == 966 {
+			req.Network = "polygon"
+		} else {
+			return "", 0, fmt.Errorf("unsupported network token: %s", token.Name)
+		}
+	}
+	endpoint := fmt.Sprintf("v2/accounts/%s/transactions", acctID)
```

Comment:
I see there are potentially newer api docs at https://docs.cdp.coinbase.com/exchange/reference/exchangerestapi_postwithdrawcrypto

Reply:
Same comment for this as below.. this is the Coinbase Exchange API, not the Advanced Trade API.